### PR TITLE
[FS] Remove unnecessary / inefficient code

### PR DESCRIPTION
**Pull request by quantumagi**
This code appears to be looking up a `TransactionData` that it already has, using the keys from the same `TransactionData`..

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs [hunk 1]
```diff
@@ -486,20 +486,16 @@ private bool RemoveTransaction(Transaction transaction)
...
                 {
                     continue;
                 }
 
                 // Get the transaction being spent and unspend it.
-                TransactionData spentTransaction = this.Wallet.MultiSigAddress.Transactions.SingleOrDefault(t => (t.Id == tTx.Id) && (t.Index == tTx.Index));
-                if (spentTransaction != null)
-                {
-                    this.logger.LogTrace("Unspending {0}-{1}", spentTransaction.Id, spentTransaction.Index);
+                this.logger.LogTrace("Unspending {0}-{1}", spentTransaction.Id, spentTransaction.Index);
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: OK but above we get it from the outpointLookup and below from the MultiSigAddress transaction collection?

Comment (fassadlr) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: I.e. in the previous code we first check `outpointLookup` and then we check `this.Wallet.MultiSigAddress.Transactions` ?

Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: See this:

`private Dictionary<OutPoint, TransactionData> outpointLookup => this.Wallet.MultiSigAddress.Transactions.GetOutpointLookup();`

Comment (fassadlr) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: Ah ok yes I see now :)

**Pull request merged**